File: opencv_codes/open_copy.py
#!/usr/bin/env python

import numpy as np

import cv2

image_name = "flower"

print 'read an image from file'
img = cv2.imread("Images/flower.jpeg")

print 'create a  window to display the image'
cv2.namedWindow("Image", cv2.WINDOW_NORMAL)

print 'display the image'
cv2.imshow("Image", img)

print 'press a key in the keyboard to make a copy'
cv2.waitKey(0)

print 'Image copied to folder Images/copy'
cv2.imwrite("Images/Copy/flower-copy.jpeg", img)

File: opencv_codes/open_copy.cpp
#include <highgui.h>
#include <opencv2/imgproc/imgproc.hpp>
#include <stdio.h>
#include "opencv2/opencv.hpp"

using namespace cv;
using namespace std;

int main() {

    Mat image;

    image = imread("/home/catkin_ws/src/beginner_tutorials/opecv_codes/Images/blackwhite.jpg", CV_LOAD_IMAGE_COLOR);

    if(!image.data)
    {
        cout<<"Could not open or find the image"<<endl;
        return -1;
    }
    
    namedWindow("Window", CV_WINDOW_AUTOSIZE);
    imshow("Window", image);

    imwrite("/home/catkin_ws/src/beginner_tutorials/opecv_codes/Images/Copy/copy.jpg", image);

    waitKey(0);
    return 0;
}